test/e2e/aws_test.go

```diff
@@ -406,6 +408,42 @@ var _ = Describe("functional tests", func() {
 		})
 	})
 
+	Describe("Cluster should be highly available", func() {
+		It("", func() {
+			By("Creating a workload cluster with single control plane")
+			setup.multipleAZ = true
+			clusterK8sClient := makeSingleControlPlaneCluster(setup)
+
+			By("Creating the second Control Plane Machine in second AZ")
+			awsMachineName := setup.cpAWSMachinePrefix + "-1"
+			bootstrapConfigName := setup.cpBootstrapConfigPrefix + "-1"
+			machineName := setup.cpMachinePrefix + "-1"
+			setup.availabilityZone = availabilityZones[1]
+			createAdditionalControlPlaneMachine(setup, machineName, awsMachineName, bootstrapConfigName)
+
+			By("Creating the MachineDeployment in second AZ")
+			setup.availabilityZone = availabilityZones[1]
+			setup.subnetId = nil
+			createMachineDeployment(setup)
+
+			By("Creating the nginx service exposed as LB")
+			_, ingressHostname := createLBService(metav1.NamespaceDefault, "test-nginx-svc", clusterK8sClient)
+
+			By("Checking if the service is accessible")
+			pingService(ingressHostname)
```

Comment:
Since we are testing Control Plane availability, it might be better to just test the ability to write to the API Server. The type of resource should not matter much, but as long as we can create a resource we know that the API Server (and etcd) are in a good state.